### PR TITLE
Rdm 12856

### PR DESCRIPTION
**Pull request by sem-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-12856

### Change description ###
Add functional tests for case disposer

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/ccd/data/dao/CaseLinkRepository.java [hunk 1]
```diff
@@ -12,4 +14,7 @@
 
     List<CaseLinkEntity> findByCaseId(final Long caseId);
 
+    @Query("FROM CaseLinkEntity WHERE linked_case_id IN :queryLinkedCases")
```
Comment (KleoG) [inline src/main/java/uk/gov/hmcts/reform/ccd/data/dao/CaseLinkRepository.java:17]: No sure if the SELECT is needed in jpa?

File: src/functionalTest/resources/application-functional.yaml [hunk 2]
```diff
@@ -6,7 +6,3 @@ elasticsearch:
   hosts: ${ELASTICSEARCH_DATA_NODES_HOSTS:"http://localhost:9200"}
   request.timeout: ${ELASTICSEARCH_REQUEST_TIMEOUT:6000}
   cases.index.name.pattern: ${ELASTICSEARCH_CASES_INDEX_NAME_PATTERN:%s_cases}
-
-deletable.case.types: ${DELETE_CASE_TYPES:FT_MasterCaseType}
```
Comment (KleoG) [inline src/functionalTest/resources/application-functional.yaml:10]: Do you no longer need deletable case types?

**Review by KleoG — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/ccd/data/dao/CaseLinkRepository.java (hunk 1 above)
Comment (sem-hmcts) [reply src/main/java/uk/gov/hmcts/reform/ccd/data/dao/CaseLinkRepository.java:17]: I believe the `SELECT` clause is inferred/implied

File: src/functionalTest/resources/application-functional.yaml (hunk 2 above)
Comment (sem-hmcts) [reply src/functionalTest/resources/application-functional.yaml:10]: We still do need it but it's injected using a different strategy - please see [here](https://github.com/hmcts/ccd-case-disposer/blob/0c6a6a57ef2c26fac629e8cba94238685b8156b0/src/functionalTest/java/uk/gov/hmcts/reform/ccd/parameter/TestParameterResolver.java#L50) and [here](https://github.com/hmcts/ccd-case-disposer/blob/RDM-12856/src/functionalTest/java/uk/gov/hmcts/reform/ccd/parameter/TestParameterResolverTest.java)
This approach enables us change the `deletable.case.types (DELETABLE_CASE_TYPES_PROPERTY in functional test context)` at runtime.

Comment (KleoG) [reply src/functionalTest/resources/application-functional.yaml:10]: 👍

File: src/main/java/uk/gov/hmcts/reform/ccd/data/dao/CaseLinkRepository.java (hunk 1 above)
Comment (KleoG) [reply src/main/java/uk/gov/hmcts/reform/ccd/data/dao/CaseLinkRepository.java:17]: 👍

**Comment by sem-hmcts** [comment]
Closing this PR as all the changes in here are also in #14

**Pull request closed without merging**